### ISSUE TITLE
Teams Refresh Issue

### ISSUE THREAD

**Issue by brianghig** [bluehalo/ngx-starter#230]
When a user creates, updates, or deletes a Team, the intended behavior is to update the user's session in order to reflect their latest roles across all teams.

Steps to reproduce:

1. As a non-admin user, create a team. Verify that you are redirected to the team's listing
2. Without refreshing the browser, click on the Team to view its details
3. Observe that you do not have privileges to delete the team or add members.
4. Refresh the browser ( causing a session refresh )
5. Observe that you now have privileges to delete the team or add members. This should have been available in step 3

However, the current implementation of calling the `authenticationService.reloadCurrentUser()` does not actually request the user's latest creds from the API. This forces users to have to refresh their browser in order to get the latest teams role information.

We likely need to switch these calls to use `sessionService.reloadSession()` that will pull the latest from the API and publish it out to all observers on the UI.

Doing this directly would create a circular dependency from Core Module > Teams Module > Core Module's Session Service, so we'll likely want to break out the Session, Authentication, and Authorization services into their own "Auth Module" that Core and Teams can import.

**Comment by jrassa** [comment]
There may be a simpler way to achieve this without splitting out the auth code to a separate module.  Currently the SessionService is provided by the CoreModule via its `providers` array.  However, the now preferred method for providing services is by using `@Injectable({ providedIn: 'root' })`.  I believe making this change should avoid the circular dependency issue.

More details available here:
https://angular.io/guide/singleton-services#using-providedin

**Comment by jrassa** [comment]
I took a closer look and did some quick testing.  I'm not sure there is an issue with a circular dependency here.  The Team Module is already using the SessionService and AuthorizationService without issue.  It is more a case of calling the wrong method (and calling it incorrectly).  Although, I think it would be a good idea to go back and update the services that are being provided in the CoreModule to use the `providedIn` option.

There are actually two other issues that are preventing it from working properly.  I created a WIP PR with the fixes. #231
